src/modules/init2.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable import/order */
```

Comment:
The main advantage of the init2 method is that it is standalone while init1 is woven in the application.

Being standalone creates the benefit of creating a custom `containers.ts` file for each platform and the ability to replace certain services when needed. This is what I meant with "non-browser" apps. E.g. React Native will not be able to load the settings from an INI file, so it requires a different implementation.

I think we can extract more and more logic from containers to services/controllers to make the application more OOP and flexible.

```ts
// containers.web.ts
container.bind(SettingsService).toSelf();

// containers.native.ts
container.bind(SettingsService).to(NativeSettingsService);
```

Reply:
So I agree that we are probably better off not putting all the settings loading into `init`, but I don't think that means we also need to use the `bind.toDynamicValue` approach. All of the boilerplate in there just makes me itchy. I think it's important that the composition of the modules for a certain configuration is simple, declarative, and straight forward. I should be able to look at the binding code and understand exactly what I'm getting.

Reply:
That's correct. The most important part is that the init/containers file works without dependencies. Regarding the integration switching mechanism, that's still open for debate. I commented on your declarative method below.

Some other patterns that could be used:

https://refactoring.guru/design-patterns/factory-method
https://refactoring.guru/design-patterns/abstract-factory

Integration-wise, we are almost doing this. Following the factory pattern, the client (controller) should request the concrete integration. The downside is that the client must know which concrete integrations are available.

```tsx
export default class AccountController {
  constructor() {
    const integration = 'jwp';
    this.checkoutService = getModule(integration === 'jwp' ? InPlayerCheckoutService : CleengCheckoutService);
```

InversifyJS has made the factory helper for this, but this will reintroduce the boilerplate in the init code again. 

The Controller can also be considered the bridge in the [Bridge Pattern](https://refactoring.guru/design-patterns/bridge), although there will always be one concrete controller.

Reply:
Luckily we don't have orthogonal permutations. We basically have the 2 integration types which determine a full set of modules plus or minus some optional ones like watchlist. I've always found the factory pattern to be over complicated for most practical code. Regardless, I think Dependency Injection / IOC takes the place of any of these patterns with this PR. We never have a case where we need to have multiple implementations of anything bound, so assuming we setup the bindings properly, the app never needs to worry about anything else like factories later on.

There's a number of ways we could make the different module compositions less coupled but I don't think it's bad to have them statically defined in code in a single init file. If we get to the point where we have 5+ options then maybe we can think about abstracting it.